File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/Intake.java
package org.firstinspires.ftc.teamcode.opmodes;

import com.qualcomm.robotcore.hardware.DcMotor;

import org.firstinspires.ftc.teamcode.internals.hardware.Devices;
import org.firstinspires.ftc.teamcode.internals.registration.OperationMode;
import org.firstinspires.ftc.teamcode.internals.registration.TeleOperation;

/**
 * This runs the intake system.
 * <p>
 * Connections: A motor in port 0 and a controller.
 * <p>
 * Controls: Press dpad up to turn on the intake.
 * <p>
 * @param <motorPower>
 */
public class Intake<motorPower> extends OperationMode implements TeleOperation {
    @Override
    public void construct() {

    }

    @Override
    public void run() {
        double motorPower = Devices.motor0.getPower();
        Devices.motor0.setPower(100);
        if (Devices.controller1.getDpadUp()) {
            if (motorPower == 100) {
                Devices.motor0.setPower(100);
            }
            if (motorPower < 100) {
                Devices.motor0.setPower(motorPower - 10);
                if (motorPower <= 0) {
                    Devices.motor0.setPower(0);
                }
            }
        }
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/Intake.kt
package org.firstinspires.ftc.teamcode.opmodes

import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.internals.templates.BaseOpMode
import org.firstinspires.ftc.teamcode.scripts.ContinuousServoScript


@TeleOp(name = "Intake", group = BaseOpMode.DEBUG_GROUP_NAME)
class Intake: BaseOpMode() {
    override fun construct() {
        addScript(ContinuousServoScript(id = 0, inverted = false, input = {((gamepad1.right_trigger - gamepad1.left_trigger).toDouble())}))
        addScript(ContinuousServoScript(id = 1, inverted = true, input = {(gamepad1.right_trigger - gamepad1.left_trigger).toDouble()}))
    }

    override fun run() {
    }

    override fun onStop() {
    }
}